### ISSUE TITLE
Deciding the future of `commonmark.py`

### ISSUE THREAD

**Issue by astrojuanlu** [rtfd/CommonMark-py#308]
Hi everyone, from Read the Docs we would like to open up the question about the future of `commonmark.py`. The landscape of Markdown in Python and Sphinx has changed a lot over the past 12 months, and in particular [markdown-it-py](https://github.com/executablebooks/markdown-it-py) by @chrisjsewell _et al_ achieves a level of extensibility that is beyond what's possible with `commonmark.py` (see https://github.com/readthedocs/commonmark.py/issues/106#issuecomment-583203795). Similarly, [MyST-Parser](https://github.com/executablebooks/MyST-Parser/) provides a broader feature set than what `recommonmark` offers.

As a result of the latter, and with the hope of optimizing efforts, a couple of months ago [we decided to deprecate recommonmark in favor of MyST-Parser](https://github.com/readthedocs/recommonmark/issues/221). Since `commonmark.py` has a similar status, we would also like to deprecate it - but given that the project has a richer history and it's also a lower level package, we would like to first ask the current maintainers their opinion.

@nikolas and @lu-zero are the ones that have been merging code to `commonmark.py` in recent years. There are a bunch of open bugs and pull requests that haven't received much attention, and the last release is from October 2019 (<sub>ah, the good old times without global pandemics</sub>). Nobody from Read the Docs plans to step in and address them, so we would like to offer two possibilities:

- We find a new GitHub organization for the project, where the same or new maintainers can give it some love, or
- We add a similar deprecation note than the one we placed on recommonmark README, and archive the repository here.

If there is no response or consensus in a reasonable time frame, the latter option wins, since anyone can fork the repository anyway. But we would like to give others the chance to speak up so the complete repository, with all the issues and pull request history, can be cleanly transferred.

Thoughts?

cc @ericholscher

**Comment by nikolas** [comment]
I'm all for transferring commonmark.py to [Jazzband](https://jazzband.co/) so it can receive better regular maintenance and attention!

**Comment by nikolas** [comment]
@astrojuanlu I've submitted a proposal to move this package to jazzband here: https://github.com/jazzband/help/issues/221

**Comment by astrojuanlu** [comment]
Thanks a lot @nikolas ! I subscribed there 👍🏽

**Comment by lu-zero** [comment]
I'm fine with moving it or favor markdown-it-py.

**Comment by ericholscher** [comment]
Yea, Jazzband seems like a good home if we want to keep it maintained in some fashion 👍

**Comment by tejasvi** [comment]
I would add that no other commonmark compliant parser in python provides source level map for the ast which is required for efficient documentation manipulation.

**Comment by astrojuanlu** [comment]
> I would add that no other commonmark compliant parser in python provides source level map for the ast which is required for efficient documentation manipulation.

Summonning @chrisjsewell on this

**Comment by chrisjsewell** [comment]
> no other commonmark compliant parser in python provides source level map ... Summonning @chrisjsewell on this

yep markdown-it/markdown-it-py does

If you want to use [commonmark.js](https://github.com/commonmark/commonmark.js) over [markdown-it](https://github.com/markdown-it/markdown-it) that obviously that's absolutely fine.
But I looked through all the markdown parsers for myst-parser, and well  there is a reason markdown-it has 10 times the GitHub stars, and is used by VS Code etc; it's just a lot better, in terms of parsing logic, performance, extensibility, source mapping, etc

**Comment by tejasvi** [comment]
@chrisjsewell Does it provide map from tokens to locations in source? AFAIK original JS markdown-it does not provide them https://github.com/markdown-it/markdown-it/issues/336 In JS world, only commonmark.js (block granularity) and remark (inline granularity) provide such info.

Edit: Oops, my memory failed me. Markdown-it does provide line position but not character position. Somehow I couldn't find them the list of token generated. In light, of that my original comment is moot.

Edit2: Thanks @chrisjswell. For me markdown-it-py is the way forward and you saved the day when I was finishing up ast generation implementation via NodeJS over rpc. (after brief adventure with [JS2Py](https://github.com/PiotrDabkowski/Js2Py) which ended up being too slow)

**Comment by chrisjsewell** [comment]
> Does it provide map from tokens to locations in source?

Yes it does, in the `map` attribute; it provides block level mapping to source lines: https://markdown-it.github.io/
But then you have already discussed this in markdown-it/markdown-it#336

No doubt, inline level (column) mapping would be great, but personally I think it would be a lot better to create an adapted version of markdown-it rather than commonmark.js, if you ever want to consider extending the syntax past commonmark in any way.

**Comment by chrisjsewell** [comment]
https://ui.toast.com/weekly-pick/en_20200402 looks interesting though, will certainly keep an eye on it 😄

**Comment by ericholscher** [comment]
Any progress on Jazzband migration here? Seems like this is mostly stalled, so tempted to add a note archiving it unless anyone else feels strongly.

**Comment by ericholscher** [comment]
I'm now going to archive this repo. Please reach out to me if anyone is interested in continuing to maintain this, but it seems there's already a solid replacement, and we aren't going to do any more work on this project.

**Comment by nikolas** [comment]
Project is now deprecated. I'm just cleaning some things up.
